query-service-impl/src/main/java/org/hypertrace/core/query/service/postgres/converters/DefaultColumnRequestConverter.java

```diff
@@ -453,28 +453,28 @@ private String convertLiteralToString(LiteralConstant literal, Builder paramsBui
     String ret = null;
     switch (value.getValueType()) {
       case STRING_ARRAY:
-        StringBuilder builder = new StringBuilder("(");
-        String delim = "";
-        for (String item : value.getStringArrayList()) {
-          builder.append(delim);
-          builder.append(QUESTION_MARK);
-          paramsBuilder.addStringParam(item);
-          delim = ", ";
-        }
-        builder.append(")");
-        ret = builder.toString();
+        ret = buildArrayValue(value.getStringArrayList(), paramsBuilder::addStringParam);
         break;
       case BYTES_ARRAY:
-        builder = new StringBuilder("(");
-        delim = "";
-        for (ByteString item : value.getBytesArrayList()) {
-          builder.append(delim);
-          builder.append(QUESTION_MARK);
-          paramsBuilder.addByteStringParam(item);
-          delim = ", ";
-        }
-        builder.append(")");
-        ret = builder.toString();
+        ret = buildArrayValue(value.getBytesArrayList(), paramsBuilder::addByteStringParam);
+        break;
+      case LONG_ARRAY:
+        ret = buildArrayValue(value.getLongArrayList(), paramsBuilder::addLongParam);
+        break;
+      case INT_ARRAY:
```

Comment:
@satish-mittal FYI: we might have to add the support for trino too. But, we also have to check if those queries works fine first.